_posts/2018/2018-05-13-a-job-should-know-whether-to-run-itself.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "A job should know whether to run itself"
+description: 'A little self.knowledge goes a long way.'
+layout: article
+category: ruby
+image:
+  base: '2018/a-job-should-know-whether-to-run-itself'
+  alt: 'A queue'
+  source: 'https://unsplash.com/photos/Nzb4LBsctyQ'
+  credit: 'Hal Gatewood'
+---
+
+Like other `ActiveWhatever` objects in Rails, a job also comes with a selection of callbacks that are run at different points in it's lifecycle.
+
+A list of the available callbacks are in the [Activejob Rails Guide](http://guides.rubyonrails.org/active_job_basics.html#callbacks).
+
+We can use these callbacks to save a lot of conditional logic for jobs that might not always need to be run.
+
+
+## Instead of…
+
+...checking whether a job should be enqueued when you enqueue it...
+
+```ruby
+class DoItLaterJob < ApplicationJob
+  def perform(thing)
+    # do work
+  end
+end
+
+if thing.should_be_done?
+  DoItLaterJob.perform_later(thing)
+end
+```
+
+
+## Use…
+
+...use an early `return` to do nothing at the point the job is executed.
+
+```ruby
+class DoItLaterJob < ApplicationJob
+  def perform(thing)
+    return unless thing.should_be_done?
+    # do work
+  end
+end
+
+# then
+DoItLaterJob.perform_later(thing)
+```
+
+
+# Or...
+
+...even throw the symbol `:abort` in a callback so a job does not even enqueue itself.
```

Comment:
Take out one of the 'even's here...I think the first one should go.

Reply:
Both.